test/functional/collection_tests.js

```diff
@@ -12,1648 +11,1040 @@ describe('Collection', function() {
+                        expect(found_does_not_exist).to.be.false;
+                        done();
+                      });
+                    }
+                  );
                 });
-
-                test.ok(found);
-                test.ok(found2);
-
-                // Let's close the db
-                client.close();
-                done();
-              });
-            });
-          });
-        });
-      });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should ensure strict access collection', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
-
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
-
-        db.collection('does-not-exist', { strict: true }, function(err) {
-          test.ok(err instanceof Error);
-          test.equal(
-            'Collection does-not-exist does not exist. Currently in strict mode.',
-            err.message
-          );
-
-          db.createCollection('test_strict_access_collection', function(err) {
-            test.equal(null, err);
-            db.collection(
-              'test_strict_access_collection',
-              self.configuration.writeConcernMax(),
-              function(err, collection) {
-                test.equal(null, err);
-                test.ok(collection.collectionName);
-                // Let's close the db
-                client.close();
-                done();
               }
             );
           });
         });
       });
-    }
-  });
+    });
 
-  /**
-   * @ignore
-   */
-  it('should perform strict create collection', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+    /**
+     * @ignore
+     */
+    it('should correctly retrieve listCollections', function(done) {
+      db.createCollection('test_collection_names', err => {
+        expect(err).to.not.exist;
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+        db.listCollections().toArray((err, documents) => {
+          let found = false;
+          let found2 = false;
 
-        db.createCollection('test_strict_create_collection', function(err, collection) {
-          test.equal(null, err);
-          test.equal('test_strict_create_collection', collection.collectionName);
+          documents.forEach(document => {
+            if (
+              document.name === this.configuration.db + '.test_collection_names' ||
```

Comment:
We should use a variable like `config` at the beginning of the test to store `this.configuration` in case the context of `this` changes throughout the test.